### PR TITLE
2022 02 help preview - proof of concept

### PR DESCRIPTION
**Pull request by aurel-l**
## Purpose
A preview page for help files

## Approach
Create a specifc "private" page at `/help/_preview`
Reuse the same component used to render help pages normally in order to preview in a similar context
Prepend a section to load the content from a URL (could add from copy/paste too if we go forward with this):
 - URL could be raw mardown file, github page for a markdown file, or github wiki page

In that section, add validation/verification:
 - view original file
 - view metadata from frontmatter:
   - check that title is here
   - check that categories are here, also preview how many other articles have this category, visual warning if 0
   - display other detected metadata but warn that they will be ignored

## Testing
Manual tests

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [ ] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

### CODE REVIEW

File: src/help/components/entry/EntryPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    }
+    setUrl(url);
+  }, []);
+
+  // Note: copy structure of "normal" help entry page in order to preview it
+  // Maybe we should just use the actual help entry page?
+  return (
+    <>
+      <HTMLHead>
+        {/* Don't index 4xx pages */}
```
Comment (dlrice) [inline src/help/components/entry/EntryPreview.tsx:127]:
```suggestion
```
i think this is from a copy/paste

File: src/help/components/entry/EntryPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        <div>The content outside of this area is a preview</div>
+        <label>
+          Load article content from a URL:{' '}
+          <input type="url" onChange={handleChange} />
+        </label>
+        {data && (
+          <output>
+            <details>
+              <summary>Raw loaded content</summary>
+              <CodeBlock>{data}</CodeBlock>
```
Comment (dlrice) [inline src/help/components/entry/EntryPreview.tsx:140]: As some of these files can be really long what do you think about doing something like:
```
max-height: 40vh;
overflow: auto;
```

File: src/help/components/entry/EntryPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                <dl>
+                  <dt>{key} ❌ ignored/invalid field</dt>
+                  <dd>{value}</dd>
+                </dl>
+              ))}
+            </details>
+          </output>
+        )}
+      </section>
+      <HelpEntry {...props} overrideContent={parsedData} />
```
Comment (dlrice) [inline src/help/components/entry/EntryPreview.tsx:178]:
```suggestion
      {parsedData !== defaultData && (
        <HelpEntry {...props} overrideContent={parsedData} />
      )}
```
Just draws the eyes more to the url on initial load

File: src/help/components/entry/EntryPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+  // Note: copy structure of "normal" help entry page in order to preview it
+  // Maybe we should just use the actual help entry page?
+  return (
+    <>
+      <HTMLHead>
+        {/* Don't index 4xx pages */}
+        <meta name="robots" content="noindex" />
+      </HTMLHead>
+      <section className={styles['private-area']}>
+        <div>The content outside of this area is a preview</div>
```
Comment (dlrice) [inline src/help/components/entry/EntryPreview.tsx:131]:
```suggestion
        <div>Content loaded in this page is a preview of what a help article will look like when rendered in the website</div>
```

File: src/help/components/entry/styles/entry-preview.module.scss [hunk 5]
```diff
@@ -0,0 +1,28 @@
+@import 'franklin-sites/src/styles/settings';
+
+.private-area {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  inline-size: 100%;
+  min-block-size: 100px;
+  padding: $global-padding;
+  background: lightcoral;
```
Comment (dlrice) [inline src/help/components/entry/styles/entry-preview.module.scss:10]: Was a little tricky to read the text with the contrast, how about a slightly lighter tone such as `peachpuff` or `moccasin`?

**Comment by dlrice** [comment]
I know this is just for internal use so it doesn't need to be highly polished but upon first usage I had to spend a little time to parse the `Parsed metadata`. Wondering if it would be slightly easier to understand put these this info in a 2 x 2 table that is always visible eg:

```
+------------+-------------------------------------+
| Title      | Chordata protein annotation project |
| Categories | Biocuration project (8), ...        |
+------------+-------------------------------------+
```

Screenshot 2022-02-03 at 15 40 11

File: src/help/components/entry/EntryPreview.tsx (hunk 2 above)
Comment (aurel-l) [reply src/help/components/entry/EntryPreview.tsx]: Changed in https://github.com/ebi-uniprot/uniprot-website/pull/539/commits/77e306f69c6cc29745cbbbf70e826b1e14908ce8

File: src/help/components/entry/styles/entry-preview.module.scss (hunk 5 above)
Comment (aurel-l) [reply src/help/components/entry/styles/entry-preview.module.scss]: Changed in https://github.com/ebi-uniprot/uniprot-website/pull/539/commits/77e306f69c6cc29745cbbbf70e826b1e14908ce8

**Comment by aurel-l** [comment]
> I know this is just for internal use so it doesn't need to be highly polished but upon first usage I had to spend a little time to parse the `Parsed metadata`. Wondering if it would be slightly easier to understand put these this info in a 2 x 2 table that is always visible eg:
> 
> ```
> +------------+-------------------------------------+
> | Title      | Chordata protein annotation project |
> | Categories | Biocuration project (8), ...        |
> +------------+-------------------------------------+
> ```
> 
> Screenshot 2022-02-03 at 15 40 11

Changed in https://github.com/ebi-uniprot/uniprot-website/pull/539/commits/77e306f69c6cc29745cbbbf70e826b1e14908ce8

Moved stuffed around, and styled the description list like a table

Also added a feature in order to preload a URL (from the history state) if the page is reloaded or when going back to the page after navigating away.

**Review by dlrice — APPROVED** [review]

**Pull request merged**